src/derivative_utils.jl

```diff
@@ -396,5 +411,73 @@ function update_W!(nlsolver::NLSolver, integrator, cache::OrdinaryDiffEqConstant
   nothing
 end
 
+indexed_update_W!(integrator, cache, dt, idx, repeat_step) = 
+  indexed_update_W!(cache.nlsolver[idx], integrator, cache, dt, idx, repeat_step)
+
+function indexed_update_W!(nlsolver::NLSolver, integrator, cache::OrdinaryDiffEqMutableCache, dt, idx, repeat_step)
+  if isnewton(nlsolver)
+    indexed_calc_W!(nlsolver, integrator, cache, dt, idx, repeat_step, true)
+  end
+  nothing
+end
+
+function indexed_calc_W!(nlsolver, integrator, cache::OrdinaryDiffEqMutableCache, dtgamma, idx, repeat_step, W_transform=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack J,W = cache
```

Comment:
if you just had the `nlsolver` use the `W` within it, it should be fine without having a whole new function?

Reply:
Yup it should be, but it's a workaround while W and J are in alg cache. I will remove them from these indexed versions.

Reply:
Ah we don't have a J in NLNewtonCache right now. That would be a breaking and big change